cli/demo_sidechain.sh

```diff
@@ -0,0 +1,143 @@
+#   integritee-service run
+#
+# then run this script
+
+# usage:
+#  export RUST_LOG_LOG=integritee-cli=info,ita_stf=info
+#  demo_sidechain.sh -p <NODEPORT> -A <WORKER1PORT> -B <WORKER2PORT> -m file
+#
+# TEST_BALANCE_RUN is either "first" or "second"
+# if -m file is set, the mrenclave will be read from file
+
+while getopts ":m:p:A:B:t:" opt; do
+    case $opt in
+        t)
+            TEST=$OPTARG
+            ;;
+        m)
+            READMRENCLAVE=$OPTARG
+            ;;
+        p)
+            NPORT=$OPTARG
+            ;;
+        A)
+            WORKER1PORT=$OPTARG
+            ;;
+        B)
+            WORKER2PORT=$OPTARG
+            ;;
+    esac
+done
+
+# using default port if none given as arguments
+NPORT=${NPORT:-9944}
+WORKER1PORT=${WORKER1PORT:-2000}
+WORKER2PORT=${WORKER2PORT:-3000}
+
+echo "Using node-port ${NPORT}"
+echo "Using trusted-worker-1-port ${WORKER1PORT}"
+echo "Using trusted-worker-2-port ${WORKER2PORT}"
+
+AMOUNTSHIELD=50000000000
+AMOUNTTRANSFER=20000000000
+
+CLIENTWORKER1="./../bin/integritee-cli -p ${NPORT} -P ${WORKER1PORT}"
+CLIENTWORKER2="./../bin/integritee-cli -p ${NPORT} -P ${WORKER2PORT}"
+
+if [ "$READMRENCLAVE" = "file" ]
+then
+    read MRENCLAVE <<< $(cat ~/mrenclave.b58)
+    echo "Reading MRENCLAVE from file: ${MRENCLAVE}"
+else
+    # this will always take the first MRENCLAVE found in the registry !!
+    read MRENCLAVE <<< $($CLIENTWORKER1 list-workers | awk '/  MRENCLAVE: / { print $2; exit }')
+    echo "Reading MRENCLAVE from worker list: ${MRENCLAVE}"
+fi
+[[ -z $MRENCLAVE ]] && { echo "MRENCLAVE is empty. cannot continue" ; exit 1; }
+
+echo ""
+echo "* Create a new incognito account for Alice"
+ICGACCOUNTALICE=//AliceIncognito
+echo "  Alice's incognito account = ${ICGACCOUNTALICE}"
+echo ""
+
+echo "* Create a new incognito account for Bob"
+ICGACCOUNTBOB=//BobIncognito
+echo "  Bob's incognito account = ${ICGACCOUNTBOB}"
+echo ""
+
+echo "* Issue ${AMOUNTSHIELD} tokens to Alice's incognito account (on worker 1)"
+${CLIENTWORKER1} trusted set-balance ${ICGACCOUNTALICE} ${AMOUNTSHIELD} --mrenclave ${MRENCLAVE} --direct
+echo ""
+
+echo "* Waiting 2 seconds"
+sleep 2
+echo ""
+
+echo "Get balance of Alice's incognito account (on worker 1)"
+${CLIENTWORKER1} trusted balance ${ICGACCOUNTALICE} --mrenclave ${MRENCLAVE}
+echo ""
+
+# Send funds from Alice to Bobs account, on worker 1
+echo "* Send ${AMOUNTTRANSFER} funds from Alice's incognito account to Bob's incognito account (on worker 1)"
+$CLIENTWORKER1 trusted transfer ${ICGACCOUNTALICE} ${ICGACCOUNTBOB} ${AMOUNTTRANSFER} --mrenclave ${MRENCLAVE} --direct
+echo ""
+
+echo "* Waiting 2 seconds"
+sleep 2
+echo ""
+
+# Send funds from Alice to Bobs account, on worker 2
+echo "* Send ${AMOUNTTRANSFER} funds from Alice's incognito account to Bob's incognito account (on worker 2)"
+$CLIENTWORKER2 trusted transfer ${ICGACCOUNTALICE} ${ICGACCOUNTBOB} ${AMOUNTTRANSFER} --mrenclave ${MRENCLAVE} --direct
+echo ""
+
```

Comment:
Demo script for the sidechain. We do direct invocation call on both workers and then check the balances on worker 2. We need some sleeps in the scripts now, because the state needs more time to get updated (a specific worker is now only producing blocks every 2nd slot). So we need to wait a bit before we execute the getters.